File: backend/src/main/kotlin/com/unsam/scholarium/model/Material.kt
package com.unsam.scholarium.model

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.util.*

enum class TipoMaterial {
    APUNTE,
    PARCIAL,
    FINAL,
    GUIA_EJERCICIOS,
    OTRO,
}


enum class EstadoMaterial {
    PENDIENTE,
    PUBLICADO,
    RECHAZADO
}

@Entity
class Material(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,

    @Column(nullable = false)
    var nombre: String,

    @Column(nullable = false, length = 255)
    var descripcion: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var tipo: TipoMaterial,

    @Column(nullable = false)
    var url: String,

    @Column(nullable = false)
    var tamanio: Long,

    @Column(nullable = false)
    var tipoArchivo: String,

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var estado: EstadoMaterial = EstadoMaterial.PENDIENTE,

    @Column(nullable = true)
    var motivoRechazo: String? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "materia_id", nullable = false)
    var materia: Materia,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploadedById", nullable = false)
    var usuario: Usuario,

    @CreationTimestamp
    @Column(name = "created_at", nullable = false)
    var createdAt: Date = Date(),

    @UpdateTimestamp
    @Column(name = "updated_at")
    var updatedAt: Date? = null,
) {

}

File: frontend/src/main/types/Material/Material.ts
export type TipoMaterial =
  | 'APUNTE' | 'PARCIAL' | 'FINAL' | 'GUIA_EJERCICIOS' | 'OTRO';

export type EstadoMaterial = 'PENDIENTE' | 'PUBLICADO' | 'RECHAZADO';

export const TIPO_MATERIAL_OPCIONES: { value: TipoMaterial; label: string }[] = [
  { value: 'APUNTE', label: 'Apunte' },
  { value: 'PARCIAL', label: 'Parcial' },
  { value: 'FINAL', label: 'Final' },
  { value: 'GUIA_EJERCICIOS', label: 'Guía de ejercicios' },
  { value: 'OTRO', label: 'Otro' },
];

export interface SubirMaterialRequest {
  archivo: File;
  nombre: string;
  descripcion?: string;
  tipo: TipoMaterial;
}

export interface MaterialResponse {
  id: string;
  nombre: string;
  descripcion: string;
  tipo: TipoMaterial;
  estado: EstadoMaterial;
  url: string;
  tamanio: number;
  tipoArchivo: string;
  uploadedByEmail: string;
  createdAt: string;
  updatedAt: string | null;
}